source/Halibut/Transport/SecureWebSocketListener.cs

```diff
@@ -17,31 +17,32 @@ public class SecureWebSocketListener : IDisposable
         readonly X509Certificate2 serverCertificate;
         readonly Func<MessageExchangeProtocol, Task> protocolHandler;
         readonly Predicate<string> verifyClientThumbprint;
+        readonly Action<string, string> unauthorizedClientConnect;
         readonly ILogFactory logFactory;
         readonly Func<string> getFriendlyHtmlPageContent;
         readonly Func<Dictionary<string, string>> getFriendlyHtmlPageHeaders;
         readonly CancellationTokenSource cts = new CancellationTokenSource();
         ILog log;
         HttpListener listener;
 
-        public SecureWebSocketListener(string endPoint, X509Certificate2 serverCertificate, Action<MessageExchangeProtocol> protocolHandler, Predicate<string> verifyClientThumbprint, ILogFactory logFactory, Func<string> getFriendlyHtmlPageContent)
-            : this(endPoint, serverCertificate, h => Task.Run(() => protocolHandler(h)), verifyClientThumbprint, logFactory, getFriendlyHtmlPageContent, () => new Dictionary<string, string>())
+        public SecureWebSocketListener(string endPoint, X509Certificate2 serverCertificate, Action<MessageExchangeProtocol> protocolHandler, Predicate<string> verifyClientThumbprint, Action<string, string> unauthorizedClientConnect, ILogFactory logFactory, Func<string> getFriendlyHtmlPageContent)
```

Comment:
Can we add a new constructor instead and the rest just pass an empty callback? Then we don't need to bump a major and the other consumers of this lib won't have to change their code.

Reply:
Done.